src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -1461,42 +1442,26 @@ private GridCoverageReader getGridCoverageReader(CoverageStoreInfo info, Coverag
             throw new IOException("Could not find the raster plugin for format " + info.getType());
         }
         
-        // look into the cache
-        GridCoverageReader reader = null;
-        Object key;
-        if ( hints != null && info.getId() != null) {
-            // expand the hints if necessary
-            final String formatName = gridFormat.getName();
-            if (formatName.equalsIgnoreCase(IMAGE_MOSAIC) || formatName.equalsIgnoreCase(IMAGE_PYRAMID)){
-                if (coverageExecutor != null){
-                    if (hints != null) {
-                        // do not modify the caller hints
-                        hints = new Hints(hints);
-                        hints.add(new RenderingHints(Hints.EXECUTOR_SERVICE, coverageExecutor));
-                    } else {
-                        hints = new Hints(new RenderingHints(Hints.EXECUTOR_SERVICE, coverageExecutor));
-                    }
-                }
-            }
-            
-            key = new CoverageHintReaderKey(info.getId(), hints);
-            reader = hintCoverageReaderCache.get( key );
+        // expand the hints if necessary
+        if (hints != null) {
+            // do not modify the caller hints
+            hints = new Hints(hints);
+            hints.add(new RenderingHints(Hints.REPOSITORY, repository));
```

Comment:
Not really, the 1451 creates a new hints object in case it's null. So I would have to leave it there, but replace it with the creation of empty hints. Line count is the same, but I guess having the add out of the conditional makes it clearer that we are going to add a repository anyways.